### DISCUSSION TITLE
Problem with using LSP format feature

### DISCUSSION THREAD

**Discussion by georgejean** [NvChad/NvChad#2127] [Q&A] [answered]
Hi !
I want to use the format feature which comes with [texlab](https://github.com/latex-lsp/texlab). Here is my problem : the same config works on Windows (code is formatted by `<leader>fm`) but fails on WSL2 (Ubuntu 22.04.2 LTS). I really wish someone could test my config on a pure linux machine and check if the problem persists. I'm not sure if it is a problem with nvChad or with texlab that's why I put my request in discussion.

I've removed all extra configs to keep the essentials. Here is my  `plugins.lua` file :
```lua
-- custom/plugins.lua
local plugins = {
  {
    "williamboman/mason.nvim",
    opts = {
      ensure_installed = {
        "texlab"
      },
    },
  },
  {
    "neovim/nvim-lspconfig",
    config = function()
      require "plugins.configs.lspconfig"
      require "custom.configs.lspconfig"
    end,
  },
  {
    "lervag/vimtex",
    lazy = false,
    init = function()
      -- Live compilation
      vim.g.vimtex_compiler_latexmk = {
        out_dir = ".out",
        options = {
          "-shell-escape",
          "-verbose",
          "-file-line-error",
          "-interaction=nonstopmode",
          "-synctex=1",
        },
      }
      vim.g.vimtex_view_method = "zathura"
    end,
  },
}
return plugins
```
and here is my `lspconfig.lua` file :
```lua
-- custom/configs/lspconfig.lua
local capabilities = require("plugins.configs.lspconfig").capabilities
local lspconfig = require "lspconfig"
local util = require "lspconfig.util"

-- enable formatting : It works on Windows, not on WSL2 :
local on_attach = function(client, bufnr)
  require("plugins.configs.lspconfig").on_attach(client, bufnr)
  client.server_capabilities.documentFormattingProvider = true
end

lspconfig["texlab"].setup {
  on_attach = on_attach,
  capabilities = capabilities,
  root_dir = function(filename)
    return util.path.dirname(filename)
  end,
  settings = {
    texlab = {
      chktex = {
        onEdit = true,
      },
      latexFormatter = "latexindent", -- comes with the defaults
    },
  },
}
```
now, when I test in a `tex` file like this one below, I can't format with WSL2, but it works on Windows (with another PDF Viewer : it's my only change in the config) :
```tex
% try.tex
\documentclass{minimal}
\begin{document}
hello men !!

how are you?

Fine, thanks ! And you?


     \begin{equation}  % bad indent not fixed by <leader>fm
  a = b^2 \ \ \frac{3}{4} \ \ x^3=x \leftrightarrow x^2 = 1
\end{equation}

\begin{equation}
  \int_{3}^{9}(x-6)^5 dx
\end{equation}
Hey!! hello man!

Au revoir!
\end{document}
```
Usually, I use `null_ls` for formatting but it should also work here... Could somebody test it and give me some clues?

**Comment by georgejean** [comment]
One detail : I use it in my `init.lua`:
```lua
-- custom/init.lua
local g = vim.g
g.maplocalleader = " "
```

**Reply by siduck** [reply]
@georgejean does formatting work when you comment those formatting related lines in the default lspconfig's on_attach function?

**Reply by georgejean** [reply]
No, there is no more success...I really can't figure why.
![vimtex5](https://github.com/NvChad/NvChad/assets/11246966/9ed07a34-5045-4a65-ab8e-476ba42026b4)

I've also removed the Vimtex config, without success.

**Reply by siduck** [reply]
@georgejean  reopen neovim to see the changes

**Reply by georgejean** [reply, marked as answer]
@siduck
Ok, It seems that it comes from `latexindent` which is not correctly installed (I've installed a full [TeXLive](https://www.tug.org/texlive/quickinstall.html) though...).
![image](https://github.com/NvChad/NvChad/assets/11246966/ac5a3de6-e5eb-4286-8f6b-fc6170c74b0c)

It's undoubtedly the cause. I will investigate this.
Sorry for this false issue and the time you've spent for me.

EDIT : I confirm that after following the instructions [here](https://newptcai.gitlab.io/blog/latexindent/), the problem has gone away.
